app/src/main/java/org/stacks/app/domain/Login.kt

```diff
@@ -49,5 +61,42 @@ class Login
                 it
             }
 
+    private suspend fun generateWalletKeysFromMnemonicWords(seedPhrase: String): ExtendedKey =
+        withContext(Dispatchers.IO) {
+            val words = MnemonicWords(seedPhrase)
+            val identity = BlockstackIdentity(words.toSeed().toKey("m/"))
+
+            val key = identity.identityKeys.generateChildKey(BIP44Element(true, 45))
+
+            Timber.i("Address key: ${key.keyPair.toBtcAddress()}")
+            Timber.i("Private key: ${key.keyPair.privateKey.key.toHexStringNoPrefix()}")
+            Timber.i("Public key: ${key.keyPair.publicKey.key.toHexStringNoPrefix()}")
+
+            return@withContext key
+        }
+
+     /**
+     * Gets the Identities present in the decoded cipher param, decoded cipher such as the Wallet-Config.json
+     *
+     * @param decodedCipher String - identities in JSON format
+     *
+     * @throws IdentitiesParsingFailed if it fails to convert the identities
+     */
+    private fun identitiesFromDecodeCipher(decodedCipher: String): List<IdentityModel> {
+        val listOfIdentities = mutableListOf<IdentityModel>()
+
+        try {
+            val identities = JSONObject(decodedCipher).getJSONArray("identities")
+
+            for (i in 0 until identities.length()) {
+                listOfIdentities.add(IdentityModel(identities.getJSONObject(i)))
+            }
```

Comment:
Since you need to do this JSONArray to List operation in more than one place, you might want to create an helper method `JSONArray.map {}`.